src/functions/interactivity/suggestion_router.ts

```diff
@@ -0,0 +1,189 @@
+import {
+  ParameterSetDefinition,
+  PossibleParameterKeys,
+} from "../../parameters/mod.ts";
+import { SlackFunctionDefinition } from "../mod.ts";
+import { UnhandledEventError } from "../unhandled-event-error.ts";
+import { enrichContext } from "../enrich-context.ts";
+import { FunctionDefinitionArgs, FunctionRuntimeParameters } from "../types.ts";
+import type {
+  BlockActionConstraint,
+  BlockSuggestionHandler,
+  RuntimeSuggestionContext,
+  SuggestionContext,
+} from "./types.ts";
+import { BlockAction } from "./block_kit_types.ts";
+import {
+  matchBasicConstraintField,
+  normalizeConstraintToArray,
+} from "./matchers.ts";
+
+/**
+ * Define a suggestion "router" and its input and output parameters for use in a Slack application. The SuggestionRouter will route incoming action events to action-specific handlers.
+ * @param {SlackFunctionDefinition<InputParameters, OutputParameters, RequiredInput, RequiredOutput>} func Reference to your previously-created SlackFunction, defined via DefineFunction
+ * @returns {SuggestionRouter}
+ */
+export const BlockSuggestionRouter = <
```

Comment:
This isn't exported from `deno-slack-sdk/mod.ts`, but `BlockActionsRouter` is. I bring this up for
1. Consistency
2. The jsdoc comments here are pretty extensive, but I'm not sure they're being surfaced anywhere (I'm using `SlackFunction().addBlockSuggestionHandler()` but we don't surface this directly.

What do you think? Should we export from the top level or alternatively should we deprecate `BlockActionsRouter` being exported from there?

Reply:
Good callout. So I have a [PR up](https://slack-github.com/slack/webapp/pull/158071) to update the interactivity docs on api.slack.com to not use these Router constructs but rather the `add*Handler` methods hanging off the top-level function. Once that is merged, I think we can then go through the motions to deprecate / no longer export the Router bits and, probably at that time it would be appropriate, also refactor this entire mess to DRY up the interactivity routing for all interactivity events.